src/ngraph/op/add.cpp

```diff
@@ -33,8 +33,10 @@ shared_ptr<Node> op::Add::copy_with_new_args(const NodeVector& new_args) const
     return make_shared<Add>(new_args.at(0), new_args.at(1));
 }
 
```

Comment:
What if there were an overloading of `genearate_adjoints` that took a `Node` as its argument, and the vector'd `generate_adjoints` default implementation called it with the `deltas.at(0)`. Then you wouldn't need to change all these.